tcl/dialog_audio.tcl

```diff
@@ -169,19 +185,30 @@ proc ::dialog_audio::pdtk_audio_dialog {mytoplevel \
         $mytoplevel.settings.bsc.rate2 \
         $mytoplevel.settings.bsc.rate3 \
          -side left
+    if {$audio_isfixedsr == 1} {
+        $mytoplevel.settings.srd.sr_entry config -state "disabled"
+        $mytoplevel.settings.bsc.rate1 config -state "disabled"
+        $mytoplevel.settings.bsc.rate2 config -state "disabled"
+        $mytoplevel.settings.bsc.rate3 config -state "disabled"
+    }
+
     label $mytoplevel.settings.bsc.bs_label -text [_ "Block size:"]
     set blocksizes {64 128 256 512 1024 2048}
     set bsmenu \
         [eval tk_optionMenu $mytoplevel.settings.bsc.bs_popup audio_blocksize $blocksizes]
-
     pack $mytoplevel.settings.bsc.bs_popup -side right
     pack $mytoplevel.settings.bsc.bs_label -side right -padx {0 10}
-    if {$audio_callback >= 0} {
```

Comment:
I think this still needs to be checked for as I now see the "Use callbacks" checkbox on macOS when it didn't show in the last 0.51-4 release.

Reply:
i see this as a feature ;-) : have a consistent layout of the dialog, regardless of the OS and backend.

compare this:
> A: please check the checkbox FOO
> B: hmm, the FOO checkbox is disabled

against this:
> A: please check the checkbox FOO
> B: there's no FOO checkbox
> A: sure there is; just above the BAR field
> B: no there isn't. check my screenshot
> A: ah, you are on macOS and you are using PortAudio

anyhow: i'm not particularily opinionated about this.
if it bothers you much, feel free to re-introduce the checkbox hiding (using `$audio_isfixedcallback` instead of `$audio_callback`)

Reply:
I didn't introduce the change, I believe Miller did, but I appreciate that it's no longer there and doesn't take space in the layout when it's not relevant. Now that we can disable things, *maybe* it can stay but this would a flip/flop from the last version.

Reply:
I'll re-remove it for now.